.github/workflows/sync_from_airtable.yml

```diff
@@ -0,0 +1,330 @@
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v6
+        env:
+          DATA: ${{ fromJSON(env.DATA) }}
+          NUMBER: ${{env.NUMBER}}
+          OWNER: ${{env.OWNER}}
+          REPO: ${{env.REPO}}
+        with:
+          script: |
+            const { DATA, NUMBER, OWNER, REPO } = process.env;
+            const { milestone } = JSON.parse(DATA);
+            const repoName = `${OWNER}/${REPO}`;
+            try {
+                if (milestone) {
+                    const { repo, milestone_number } = milestone;
+                    if (repoName === repo) {
+                        await github.rest.issues.update({
+                            owner: OWNER,
+                            repo: REPO,
+                            issue_number: NUMBER,
+                            milestone: milestone_number || null,
+                        });
+                        core.info(`Issue ${ NUMBER }'s Milestone updated to ${ milestone_number }`)
+                    } else {
+                        throw new Error(`This milestone only exists in ${milestone.repo}.`)
+                    }
+                } else {
+                    await github.rest.issues.update({
+                        owner: OWNER,
+                        repo: REPO,
+                        issue_number: NUMBER,
+                        milestone: null,
+                    });
+                    core.info(`Issue ${ NUMBER }'s Milestone updated to 'null'`)
+                }
+            } catch (error) {
+                core.info(error)
+            }
+
+  open_closed:
+    name: Opened, Reopened, or Closed
+    needs: existing_issue
+    if: (fromJSON(needs.existing_issue.outputs.github_issue).status == 200 ) && (github.event.client_payload.action == 'opened/closed')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v6
+        env:
+          DATA: ${{ fromJSON(env.DATA) }}
+          NUMBER: ${{env.NUMBER}}
+          OWNER: ${{env.OWNER}}
+          REPO: ${{env.REPO}}
+        with:
+          script: |
+            const { DATA, NUMBER, OWNER, REPO } = process.env;
+            const { state } = JSON.parse(DATA);
+
+            await github.rest.issues.update({
+              owner: OWNER,
+              repo: REPO,
+              issue_number: NUMBER,
+              state,
+            });
+
+  edited:
+    name: Edited
+    needs: existing_issue
+    if: (fromJSON(needs.existing_issue.outputs.github_issue).status == 200 ) && (github.event.client_payload.action == 'edited')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v6
+        env:
+          DATA: ${{ fromJSON(env.DATA) }}
+          NUMBER: ${{env.NUMBER}}
+          OWNER: ${{env.OWNER}}
+          REPO: ${{env.REPO}}
+        with:
+          script: |
+            const { DATA, NUMBER, OWNER, REPO } = process.env;
+            const data = JSON.parse(DATA);
+
+            await github.rest.issues.update({
```

Comment:
I think `update` run into the same timing issue as #5851, which was tricky to troubleshoot. Something to keep an eye on.

Reply:
In this case, running `update` could overwrite edits to the issue body, title, labels, etc., which would be frustrating for a user. It might be worth looking into a different way to do this that specifically uses the add/remove methods instead of changing the whole issue to match the data provided in the request.

Reply:
Edited only runs when the Title or Description have changed. The history of both are logged in the GH issue.

Reply:
If the title/description is changed in Airtable (by us) and GH (most likely the by user, but possibly us) at the same time then the changes to GH will be overwritten. It will be in the history but I don't check the history when working on an issue so I might miss some info. I'm fine with leaving it as is for now, but it's something to keep an eye on.